mod/consensus/pkg/cometbft/service/service.go

```diff
@@ -72,8 +72,11 @@ type Service[
 	finalizeBlockState   *state
 	interBlockCache      storetypes.MultiStorePersistentCache
 	paramStore           *params.ConsensusParamsStore
-	initialHeight        int64
-	minRetainBlocks      uint64
+
+	// initialHeight is used to determine if we are
+	// proposing or processing the first block or not.
+	initialHeight   int64
```

Comment:
I like an explanation of attributes where we declare them. Happy to revert if we don't agree on this (or it breaks codebase conventions)